llvm/lib/Transforms/Scalar/InferAddressSpaces.cpp

```diff
@@ -842,8 +846,9 @@ unsigned InferAddressSpacesImpl::joinAddressSpaces(unsigned AS1,
   return (AS1 == AS2) ? AS1 : FlatAddrSpace;
 }
 
-bool InferAddressSpacesImpl::run(Function &F) {
-  DL = &F.getDataLayout();
+bool InferAddressSpacesImpl::run(Function &F_) {
+  F = &F_;
```

Comment:
Not sure I like the trailing _. Maybe use something like `CurFn` instead?

Reply:
this convention is followed in maybe half of places? Could also just move setting F to the constructor but that just moves the name collision (to a less bad place, I guess)

Reply:
Using something like `F(&F)` in the constructor list is well defined in C++ and wouldn't be a collision.

In any case, I wasn't aware this was a convention and I'm not going to block anything over this :)